src/plugins/chessevent/data/chessevent_field_reader.py

```diff
@@ -0,0 +1,34 @@
+from typing import Type, Union, Optional, cast
+from enum import IntEnum
+
+class ChessEventFieldReader:
+    def __init__(self, data: dict[str, Union[bool, str, int, float, dict[int, float], None]]):
+        self._data = data
+        self.last_key: Optional[str] = None
+
+    def get[T](self, key: str, expected_type: Union[Type[T], tuple[Type, ...]], default: Optional[T] = None) -> T:
+        self.last_key = key
+        value = self._data.get(key, default)
+
+        if value is None:
+            if default is not None:
+                return default
+            raise KeyError(f"Missing required key: {key}")
+
+        if not isinstance(value, expected_type):
+            raise TypeError(f"Expected {expected_type} for key '{key}', got {type(value).__name__}")
```

Comment:
You can access the type of a value by either `type(value)` or the `__class__` attribute.
I think I'd recommend introspection, but you do you
```suggestion
            raise TypeError(f"Expected {expected_type} for key '{key}', got {value.__class__.__name__}")
```